crates/core/component/distributions/src/component.rs

```diff
@@ -54,12 +54,32 @@ impl Component for Distributions {
             .expect("failed to retrieve current epoch from state");
         let epoch_index = current_epoch.index;
 
-        let increase: Amount = state
+        let params = state
             .get_distributions_params()
             .await
-            .expect("distribution parameters should be available")
-            .liquidity_tournament_incentive_per_block
-            .into();
+            .expect("distribution parameters should be available");
+        let increase = Amount::from(params.liquidity_tournament_incentive_per_block);
+        let current_block = u64::try_from(end_block.height).unwrap_or_default();
+        let end_block = params
+            .liquidity_tournament_end_block
+            .map_or(0u64, |b| b.get());
+        if current_block >= end_block {
```

Comment:
I think we need to add a predicate that `end_block != 0` right?

Reply:
Good catch! The intent was end_block = None => no limit, to preserve current behavior.